File: 0501_0600/0501/0501.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    s = [[0 for j in range(100)] for i in range(100)]
    for c in range(n):
        x1, y1, x2, y2 = (int(x) for x in f.readline().split())
        for i in range(min(y1, y2), max(y1, y2)):
            for j in range(min(x1, x2), max(x1, x2)):
                s[i][j] += 1
    x1, y1, x2, y2 = (int(x) for x in f.readline().split())
    q.write(str(sum(s[i][j] for j in range(min(x1, x2), max(x1, x2)) for i in range(min(y1, y2), max(y1, y2)))))

File: 0501_0600/0501/0501.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, k = 0, m[100][100], x1, x2, y1, y2;

	for (int i = 0; i < 100; ++i)
		for (int j = 0; j < 100; ++j)
			m[i][j] = 0;
	fscanf(f, "%d", &n);
	while (n--)
	{
		fscanf(f, "%d%d%d%d", &x1, &y1, &x2, &y2);
		if (x1 > x2)
			x1 ^= x2, x2 ^= x1, x1 ^= x2;
		if (y1 > y2)
			y1 ^= y2, y2 ^= y1, y1 ^= y2;
		for (; y1 < y2; ++y1)
			for (int i = x1; i < x2; ++i)
				++m[y1][i];
	}
	fscanf(f, "%d%d%d%d", &x1, &y1, &x2, &y2);
	if (x1 > x2)
		x1 ^= x2, x2 ^= x1, x1 ^= x2;
	if (y1 > y2)
		y1 ^= y2, y2 ^= y1, y1 ^= y2;
	for (; y1 < y2; ++y1)
		for (int i = x1; i < x2; ++i)
			k += m[y1][i];
	fprintf(q, "%d", k);
	return 0;
}
